Neos.Neos/Classes/View/FusionView.php

```diff
@@ -165,6 +159,12 @@ public function getFusionPath()
 
     protected function getClosestDocumentNode(Node $node): ?Node
     {
+        $nodeTypeManager = $this->contentRepositoryRegistry->get($node->contentRepositoryId)->getNodeTypeManager();
+
+        // Skip expensive subgraph lookup if the node is already a document node
+        if ($nodeTypeManager->getNodeType($node->nodeTypeName)?->isOfType(NodeTypeNameFactory::NAME_DOCUMENT)) {
```

Comment:
ideally we should not need the node type manager here because if the view is used in the controller where we only route to documents we can trust the input and not be paranoid.

(asking things the node type manager loads and initialises the configuration which is also work which is ideally not part of the rendering?)

Reply:
Yes as with all those variables, the controller should supply them and the view should complain if not. Though I think in this case the Neos FusionView accepted any kind of node so far.

I would like to fix this for 9.x if we agree on the breakinesslevel.